### PR TITLE
Tokens management for workers

### PR DESCRIPTION
**Pull request by karitra**
subj.

TODO:
 - -[ ] ~~some test samples~~
 - [x] try to make service requests async

### CODE REVIEW

File: include/cocaine/framework/worker/options.hpp [hunk 1]
```diff
@@ -25,7 +25,19 @@ namespace cocaine {
 
 namespace framework {
 
+namespace details {
+    constexpr auto DEFAULT_TOKEN_TYPE = "TVM";
+}
+
 struct options_t {
+    // taken from environment variables
+    std::string token_type;
```
Comment (3Hren) [inline include/cocaine/framework/worker/options.hpp]: Unfortunately it breaks ABI. Use `other` member with public accessors instead.

**Review by 3Hren — CHANGES_REQUESTED** [review]

File: include/cocaine/framework/worker/options.hpp [hunk 2]
```diff
@@ -25,6 +25,13 @@ namespace cocaine {
 
 namespace framework {
 
+namespace details {
+    constexpr auto TVM_TOKEN_TYPE = "TVM";
```
Comment (3Hren) [inline include/cocaine/framework/worker/options.hpp]: Move in `src`.

**Review by 3Hren — CHANGES_REQUESTED** [review]

**Comment by karitra** [comment]
Seems to work while using storage service as ticket provider mock, but don't yet have a clue how to make test cases more realistic, any advice? 

Also not sure that _async_ service call for `refresh_ticket` at ` tvm_service_impl_t::refresh_ticket_async` either correctly implemented or idiomatic. 

@3Hren, @antmat, PTAL.

File: src/worker/options.cpp [hunk 3]
```diff
@@ -21,8 +21,22 @@
...
 
+#include "defaults.hpp"
+
 using namespace cocaine::framework;
 
+namespace cocaine { namespace framework { namespace details {
+    constexpr auto DEFAULT_TOKENS_SRV_NAME = "tvm";
+    constexpr auto DEFAULT_REFRESH_INTERVAL_SEC = 15u;
+
+    constexpr auto DEFAULT_TOKEN_BODY = "very_secret";
```
Comment (3Hren) [inline src/worker/options.cpp]: Leave it empty.

File: include/cocaine/framework/worker/options.hpp [hunk 4]
```diff
@@ -41,6 +41,8 @@ struct options_t {
     std::uint32_t
     protocol() const;
 
+    const boost::any&
+    at(const std::string& name) const;
```
Comment (3Hren) [inline include/cocaine/framework/worker/options.hpp]: Closer, but not what I wanted.

Look, we have predefined number of options passed from cocaine-runtime. Therefore we can describe them explicitly:

```c++
std::string token_type() const;
std::string token_body() const;
```

**Review by 3Hren — CHANGES_REQUESTED** [review]
LGTM except those minor issues I've mentioned.

**Review by 3Hren — APPROVED** [review]

**Comment by 3Hren** [comment]
LGTM, @antmat TAL.

**Pull request merged**